api/schema/resolvers/PaymentResolver.js

```diff
@@ -77,6 +78,32 @@ module.exports = {
         deletePaymentById: (root, { id }, { models }) => {
             return models.Payment.destroy({ where: { id } })
         },
+        generateBitcoinInvoiceFromPayment: async (root, { paymentId }, { models }) => {
+            
+            const Payment = await models.Payment.findByPk(paymentId)
+            const { amount, client_id, date_paid, external_uuid, external_uuid_type } = Payment.dataValues
+
+            const client = await models.Client.findByPk(client_id)
+            
+            const isClientCurrencyBtc = client.dataValues.currency === 'BTC' || client.dataValues.currency === 'SATS'
+            if (!isClientCurrencyBtc) throw 'Client\'s currency is not Bitcoin'
+
+            const isInvoiceExpired = external_uuid && external_uuid_type === 'bitcoin' && await checkInvoiceExpired(external_uuid)
+            if (external_uuid && date_paid && !isInvoiceExpired) throw 'An active invoice already exists'
+
+            const amountInSats = client.dataValues.currency === 'BTC' ? amount * 1000000 : amount / 100;
```

Comment:
Probably want to guard this from floating point errors

```
const amountInSats = client.dataValues.currency === 'BTC'
    ? Number((amount * 1000000).toFixed(0))
    : Number((amount / 100).toFixed(0))
```

Also why are we dividing by 100 and multiplying by 1000000? 100M sats = 1 BTC

Reply:
The data type for `amount` in the database is Integer and the input for payment amount allows up to two decimal places. So the database actually stores 100 times the actual amount to store it as an Integer. These calculations are for adjusting back the amount by dividing it by 100.

Reply:
Ah okay that's right, could be worth adding a comment or describing with variables:

```
const convertedAmount = Number((amount / 100).toFixed(0))

const amountInSats = client.dataValues.currency === 'BTC'
    ? Number((convertedAmount * 100000000).toFixed(0)) // 100M sats = 1 BTC
    : convertedAmount
```